src/crimson/osd/object_context.h

```diff
@@ -35,14 +37,26 @@ struct obc_to_hoid {
   }
 };
 
+struct SnapSetContext :
+  public boost::intrusive_ref_counter<SnapSetContext,
+                                     boost::thread_unsafe_counter>
+{
+  hobject_t oid;
+  SnapSet snapset;
+  bool exists;
```

Comment:
Something like

```
/**
 * exists
 *
 * Because ObjectContext's are cached, we need to be able to express the case
 * where the object to which a cached ObjectContext refers does not exist.
 * ObjectContext's for yet-to-be-created objects are initialized with exists=false.
 * The ObjectContext for a deleted object will have exists set to false until it falls
 * out of cache (or another write recreates the object).
 */
```